### ISSUE TITLE
Design for Animation Page

### ISSUE THREAD

**Issue by britanyponvelle** [IBM-Watson/design-library#148]
Visual design for Guidelines> Animation page.

**Comment by poofichu** [comment]
@ryanbrownhill To get introduction paragraph and *maybe* animation example for this page.

**Comment by ryanbrownhill** [comment]
Content sketch:

# Animation
This is an intro paragraph the outlines IBM Motion sections and links out :neutral_face: 
## Animation Library
This is a description of the animation library that we are providing. Here is a link out of how to use this. Feel free to contribute here.

**Comment by ryanbrownhill** [comment]
@poofichu 
Here is the content - feel free to take a pass for wording / grammar.
## Animation

Motion is a fundamental element of user experience design because it demonstrates the emotional quality and vitality of the product. It gives non-verbal communication to the user and is the body language of a product. Within software, motion is conveyed through the process of animation – "the illusion of movement through a sequence of images."

Motion design enhances communication throughout the design process, from the rapid prototyping of scenarios and interfaces to delivering final coded interface animations.  The metaphor for animation comes from the IBM Design Language of metaphor elegant machine motion. The IBM Design Language Motion Section walks through the methodologies and process for creating animation for IBM software products.

## Animation Library

The animations within this library are a collection intro and exit animations for page transitions that can be utilized within our interfaces. The library was built using CSS @keyframe animations and can be implemented by using a Sass Mixin, to view more about how to use this library take a look here. Please feel free to contribute additional animations.

**Comment by kathrynmcelroy** [comment]
Animation
Motion is a fundamental element of user experience design because it demonstrates the emotional quality and vitality of the product. It gives non-verbal communication to the user and is the body language of a product. Within software, motion is conveyed through the process of animation—"the illusion of movement through a sequence of images."

Motion design enhances communication throughout the design process, from the rapid prototyping of scenarios and interfaces to delivering final coded interface animations. The metaphor for animation comes from the IBM Design Language metaphor of elegant machine motion. The IBM Design Language Motion Section walks through the methodologies and process for creating animation for IBM software products.
- Added an "em dash", moved an "of"


Animation Library
The animations within this library are a collection of intro and exit animations for page transitions that can be utilized within our interfaces. The library was built using CSS @keyframe animations and can be implemented by using a Sass Mixin. To view more about how to use this library, take a look here, and please feel free to contribute additional animations.

- added "of", separated a run-on into two sentences
- What do you mean by "intro animations"? it's not in the animation lexicon. Could we call it "entrance" or "introduction" instead?
- Do we really need to include the "@" for @keyframe? or would they understand it as just keyframe?

**Comment by ryanbrownhill** [comment]
Thanks @kathrynmcelroy updated the wiki with the changes

**Comment by poofichu** [comment]
Content added to the prototype.  @joshkimmell would you want an example animation on this page for some visual "umph" ?

http://pages.design.ibm.com/tremblay-us/ibm-watson-design-library/master/dist/animation.html

**Comment by poofichu** [comment]
@ryanbrownhill I need an example animation :)

**Comment by ryanbrownhill** [comment]
Before & After Example animation for the animation page. Will have small captions underneath explaining the reasoning for animation.

![animation-examples](https://cloud.githubusercontent.com/assets/9538395/7188909/afc51e52-e440-11e4-97b1-1eedc3fbf29e.gif)

**Comment by bobbinrobyn** [comment]
Cool, cool @ryanbrownhill

**Comment by Snugug** [comment]
Why is this blocked?

**Comment by britanyponvelle** [comment]
It was blocking Jessica because she was waiting on Ryan's examples (which he is working on today, woo!) So we can move this to "in progress"

**Comment by ryanbrownhill** [comment]
Forked @poofichu project and added the videos - going to write the captions later today.

![screen shot 2015-04-16 at 2 56 50 pm](https://cloud.githubusercontent.com/assets/9538395/7190055/f1578fb4-e448-11e4-8b1d-7077e6e20aef.png)

**Comment by ryanbrownhill** [comment]
Added Looping examples and Descriptions just need to merge it with @poofichu to pull it into the prototype.

http://pages.design.ibm.com/rgbrownh/ibm-watson-design-library/master/dist/animation.html

**Comment by Snugug** [comment]
What's the status on this?

**Comment by poofichu** [comment]
@ryanbrownhill @britanyponvelle @joshkimmell I almost feel like the videos for the animation should be in one file (if that makes sense).  If they load at different times, they are off sync a bit and there's a ton going on visually.

Also, the last section on the page (Animation Library) seems like it belongs with the UI Patterns section.

Thoughts?

**Comment by ryanbrownhill** [comment]
The animation library is just an intro on that page that links out to the documentation. (almost like a resource). As for the videos we could figure out a way to make it one but the decision for to was made because we needed it the `detail` format for the descriptions on the side.

**Comment by poofichu** [comment]
http://pages.design.ibm.com/tremblay-us/ibm-watson-design-library/master/dist/animation.html

@ryanbrownhill @britanyponvelle @Snugug as per Sam's suggestion, I put controls on the videos to make them play at the user's request instead of auto play.  I also removed the last paragraph, because it was the only page that was "go to the UI page for more" type deal.

Thoughts?

**Comment by poofichu** [comment]
@Snugug can we style the video controls in a nice, performance-minded manner?

**Comment by Snugug** [comment]
@poofichu http://blog.teamtreehouse.com/building-custom-controls-for-html5-videos, but it'd need to be progressively enhanced in (and we'd probably want a component for it)

**Comment by Snugug** [comment]
That writeup doesn't deal with progressive enhancement, so we'd need to work with that

**Comment by Snugug** [comment]
Actually, PE for that is really quite easy:

```javascript
var videos = document.querySelectorAll('video'),
    i;

for (i = 0; i < videos.length; i++) {
  videos[i].removeAttribute('controls');
  // Attach custom controls here
}
```

**Comment by Snugug** [comment]
For that specific demo, I'd also use [Modernizr's Range Input Type detect](http://v3.modernizr.com/download/#-inputtypes-cssclasses-dontmin) or similar to see if they can display a range input type. If not, don't remove basic controls.

**Comment by ryanbrownhill** [comment]
The design for this page is done, it is just getting feedback on the video controls I believe.

**Comment by Snugug** [comment]
Then we should close this issue and open a new one for styling the video controls methinks

**Comment by ryanbrownhill** [comment]
@IBM-Watson/design-acceleration can we get a thumbs up for the page design and then we will create a video controls issue. 

Here is a link to the page:
http://pages.design.ibm.com/tremblay-us/ibm-watson-design-library/master/dist/animation.html

**Comment by bobbinrobyn** [comment]
:+1:

**Comment by britanyponvelle** [comment]
:+1:

**Comment by joshkimmell** [comment]
- The controls on movie #2 don't work. 
- z-index is still off

**Comment by ryanbrownhill** [comment]
@joshkimmell what do you mean by z-index?

**Comment by joshkimmell** [comment]
if you scroll down, the video displays on top of the header

**Comment by ryanbrownhill** [comment]
Design is done & video player has been moved to a new issue. #202

Here is a link to the page design:
http://pages.design.ibm.com/tremblay-us/ibm-watson-design-library/master/dist/animation.html

**Comment by bobbinrobyn** [comment]
@Snugug How should we resolve a discrepancy in content types? This page uses the "example" content type, but the callout is NOT displayed in a list. Unfortunately, that's different than the Principles pages that use the "example" content type and display callouts in definition lists.

![screen shot 2015-04-30 at 11 17 27 am](https://cloud.githubusercontent.com/assets/9341504/7417248/7ab846f4-ef2a-11e4-9524-a106d0350488.png)
![screen shot 2015-04-30 at 11 17 37 am](https://cloud.githubusercontent.com/assets/9341504/7417249/7abb5b32-ef2a-11e4-8fcf-004e5801bb5d.png)

**Comment by Snugug** [comment]
My question would be, on the Animation page, are we using Details because of its visual layout, or is an example a better content type to use? Can we add callouts to the animations?

**Comment by bobbinrobyn** [comment]
@Snugug Fair enough, it's being styled like a detail, but the content is probably a better fit as an example. We've updated the prototype and now they're styled as examples.

**Comment by Snugug** [comment]
@bobbinrobyn This url? http://pages.design.ibm.com/tremblay-us/ibm-watson-design-library/master/dist/animation.html

**Comment by bobbinrobyn** [comment]
@Snugug Yep, that's the one.

**Comment by britanyponvelle** [comment]
:+1:

**Comment by ryanbrownhill** [comment]
:+1:

**Comment by ryanbrownhill** [comment]
Will close when #209 is merged

**Comment by Snugug** [comment]
#209 needs to be re-rolled because it wasn't made in to the `develop` branch.

@bobbinrobyn we had discussed the small `example` [template](https://github.com/IBM-Watson/design-library/blob/develop/library/templates/subcontent/_example.html) not having a description. Now that we have a description for it, that template will need to be updated. There's already code to do it in the large version, just copy it in to the small version

**Comment by bobbinrobyn** [comment]
I addressed both of these concerns – with #217 and #218

**Comment by bobbinrobyn** [comment]
Will close when #217 is merged
